src/test/assembly/asm/hexagon-types.rs

```diff
@@ -73,7 +73,7 @@ macro_rules! check_reg {
 
 // CHECK-LABEL: sym_static:
 // CHECK: InlineAsm Start
-// CHECK: r0 = #extern_static
+// CHECK: r0 = {{#+}}extern_static
```

Comment:
The `#` versus `##` syntax refers to whether the assembler should force an immediate-extender word prior to this word in the encoding.

This test change is appropriate, but I am curious about its origin so I will investigate.  Additional extender words can result in increased code size.  This test isn't intended to focus on this aspect of the emitted instructions, so I think it's appropriate as-is.

Reply:
Could it be that the assembler is reserving space for a full-size immediate that will be filled in by the linker later?